overlord/stateengine.go

```diff
@@ -68,15 +73,41 @@ func (se *StateEngine) State() *state.State {
 // must not perform long running activities during that operation, though.
 // These should be performed in properly tracked changes and tasks.
 func (se *StateEngine) Ensure() error {
+	if !se.inited {
+		for _, m := range se.managers {
```

Comment:
Good catch. I suggest having an `initialize []StateManager` field in StateEngine which is appended to by AddManager.. Ensure will then:

```
if len(se.initialize) > 0 {
        for _, manager in se.initialize {
                manager.Init(se.state)
        }
        se.managers = append(se.managers, se.initialize...)
        se.initialize = nil
}